eng/dependabot/Packages.props

```diff
@@ -1,5 +1,23 @@
 <Project>
 
+  <ItemGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
```

Comment:
Would it be any better to include this after the dependebot entries and include the condition in these instead of the dependabot references?  It feels weird to set the source-build references if the property is not set.

```
  <ItemGroup>
    <PackageReference Update="BenchmarkDotNet" Version="0.13.1" />
...
  </ItemGroup>

  <ItemGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
    <PackageReference Update="BenchmarkDotNet" Version="$(BenchmarkDotNetVersion)" Condition="'$(BenchmarkDotNetVersion)' != ''" />
  </ItemGroup>
```